lib/posix/options/pthread.c

```diff
@@ -1122,7 +1122,7 @@ int pthread_join(pthread_t pthread, void **status)
 int pthread_detach(pthread_t pthread)
 {
 	int ret = 0;
-	struct posix_thread *t;
+	struct posix_thread *t = NULL;
```

Comment:
Does this really fix it? 

If it things that `t` is uninitialized, doesn't it now just think we are de-referencing a NULL pointer in `LOG_DBG("Pthread %p detached", &t->thread);`?

Reply:
there is a null check just below.

Reply:
You mean the one on line 1129?

If the compiler understands that, then it should also understand `t = to_posix_thread(pthread);` on the line above, which sounds like being the issue here?

Reply:
Maybe, but it would fail just like if it failed in line 1128.
I'm not trying to fix this code overall. Just fix the regression that is failing CI.